make/photon/db/docker-entrypoint.sh

```diff
@@ -11,50 +11,38 @@ PGBINOLD="/usr/local/pg${PG_VERSION_OLD}/bin"
 
 PGDATAOLD=${PGDATA}/pg${PG_VERSION_OLD}
 PGDATANEW=${PGDATA}/pg${PG_VERSION_NEW}
-# to handle the PG 9.6 only
-if [ -s $PGDATA/PG_VERSION ]; then
-         PGDATAOLD=$PGDATA
-fi
-
-#
-# Init DB: $PGDATA is empty.
-# Upgrade DB: 1, has $PGDATA\PG_VERSION. 2, has pg old version directory with PG_VERSION inside.
-#
-if [ "$(ls -A $PGDATA)" ]; then
-        if [ ! -d $PGDATANEW ]; then
-                if [ ! -d $PGDATAOLD ] || [ ! -s $PGDATAOLD/PG_VERSION ]; then
-                        echo "incorrect data: $PGDATAOLD, make sure $PGDATAOLD is not empty and with PG_VERSION inside."
-                        exit 1
-                fi
-
-                initPG $PGDATANEW false
-                set +e
-                # In some cases, like helm upgrade, the postgresql may not quit cleanly.
-                # Use start & stop to clean the unexpected status. Error:
-                #   There seems to be a postmaster servicing the new cluster.
-                #   Please shutdown that postmaster and try again.
-                #   Failure, exiting
-                $PGBINOLD/pg_ctl -D "$PGDATAOLD" -w -o "-p 5433" start
-                $PGBINOLD/pg_ctl -D "$PGDATAOLD" -m fast -w stop
-                ./$CUR/upgrade.sh --old-bindir $PGBINOLD --old-datadir $PGDATAOLD --new-datadir $PGDATANEW
-                # it needs to clean the $PGDATANEW on upgrade failure
-                if [ $? -ne 0 ]; then
-                        echo "remove the $PGDATANEW after fail to upgrade"
-                        rm -rf $PGDATANEW
-                        exit 1
-                fi
-                set -e
-                echo "remove the $PGDATAOLD after upgrade success."
-                if [ "$PGDATAOLD" = "$PGDATA" ]; then
-                        find $PGDATA/* -prune ! -name pg${PG_VERSION_NEW} -exec rm -rf {} \;
-                else
-                        rm -rf $PGDATAOLD
-                fi
-        else
-                echo "no need to upgrade postgres, launch it."
+PGDATEDIR=$([ $PG_VERSION_NEW ] && echo $PGDATANEW || echo $PGDATAOLD)
+
```

Comment:
```
# we should block the upgrade path from 9.6 directly.
if [ -s $PGDATA/PG_VERSION ]; then
          echo "$error msg"
          exit 1
fi
```

and it needs test cases to cover the below patches
- 2.2 - 2.7
- 2.2 - 2.4/2.5 - 2.7